api.py

```diff
@@ -0,0 +1,92 @@
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+import os, subprocess, threading, enum
+from pytube import YouTube
+from time import sleep
+
+class State(enum.Enum):
+    INTERLUDE = "interlude"
+    PLAYING = "playing"
+
+app = FastAPI()
+current_state = State.INTERLUDE
+interlude_command = ['ffmpeg', '-re', '-stream_loop', '-1', 
+                     '-i', "./interlude.mp4", '-vf', 
+                     'scale=256:144', '-c:v', 'libx264', 
+                     '-preset', 'veryfast', '-tune', 'zerolatency', 
+                     '-c:a', 'aac', '-ar', '44100', '-f', 'flv', 
+                     'rtmp://localhost:1935/live/mystream']
+interlude_process = subprocess.Popen(interlude_command)
+video_process = None
+
+# Enable CORS
+origins = ["*"]
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+@app.get("/")
+async def root():
+    return {"message": "Hello World"}
+
+@app.get("/state/")
+async def state():
+    global current_state
+    return { "state": current_state }
+
+@app.post("/play/")
+async def play(url: str):
+    global current_state, interlude_process
```

Comment:
get rid of these `global`s across the file, instead just initializ the variables at the top of the file to `None` or some other default value